projectsmember/Pages/CooperationRequest/members.aspx.cs

```diff
@@ -17,16 +19,183 @@ protected void Page_Load(object sender, EventArgs e)
+
+                        }
+                        foreach (HtmlGenericControl span in chlng6.Controls)
+                        {
+                            string classes = span.Attributes["class"];
+                            span.Attributes["class"] = classes.Replace("placeholder", "");
+                            span.Attributes["class"] = classes.Replace("placeholder-glow", "");
+
+                        }
+                    }
+                    else if (chalengeNum == "7")
+                    {
+                        foreach (HtmlGenericControl span in chlng2.Controls)
+                        {
+                            string classes = span.Attributes["class"];
+                            span.Attributes["class"] = classes.Replace("placeholder", "");
+                            span.Attributes["class"] = classes.Replace("placeholder-glow", "");
+                        }
+                        foreach (HtmlGenericControl span in chlng3.Controls)
+                        {
+                            string classes = span.Attributes["class"];
+                            span.Attributes["class"] = classes.Replace("placeholder", "");
+                            span.Attributes["class"] = classes.Replace("placeholder-glow", "");
+
+                        }
+                        foreach (HtmlGenericControl span in chlng4.Controls)
+                        {
+                            string classes = span.Attributes["class"];
+                            span.Attributes["class"] = classes.Replace("placeholder", "");
+                            span.Attributes["class"] = classes.Replace("placeholder-glow", "");
+
+                        }
+                        foreach (HtmlGenericControl span in chlng5.Controls)
+                        {
+                            string classes = span.Attributes["class"];
+                            span.Attributes["class"] = classes.Replace("placeholder", "");
+                            span.Attributes["class"] = classes.Replace("placeholder-glow", "");
+
+                        }
+                        foreach (HtmlGenericControl span in chlng6.Controls)
+                        {
+                            string classes = span.Attributes["class"];
+                            span.Attributes["class"] = classes.Replace("placeholder", "");
+                            span.Attributes["class"] = classes.Replace("placeholder-glow", "");
+
+                        }
+                        foreach (HtmlGenericControl span in chlng7.Controls)
+                        {
+                            string classes = span.Attributes["class"];
+                            span.Attributes["class"] = classes.Replace("placeholder", "");
+                            span.Attributes["class"] = classes.Replace("placeholder-glow", "");
+
+                        }
+                    }

```

Comment:
You see there is a lot of repetition/redundancy here. 

Instead, what I would suggest is to put all those elements in a list, and then based on the challenge number, read up to that elements of the list and do these changes... 

Something like: 

```
using System.Collections.Generic;
using System.Web.UI.HtmlControls;

public void ProcessHtmlControls(int challengeNumber, List<HtmlGenericControl> htmlControls)
{
    for (int i = 0; i <= challengeNumber; i++)
    {
        if (i < htmlControls.Count)
        {
            // Process the HtmlGenericControl element at index i
            HtmlGenericControl control = htmlControls[i];
            // Do your specific processing here
            // For example, you can modify the control's properties or perform some actions

            // Example: Setting the inner text of the control to the current index
            control.InnerText = i.ToString();
        }
        else
        {
            // Handle the case when challengeNumber exceeds the number of available elements in the list
            // You can choose to throw an exception, log an error, or perform any other appropriate action
            // In this example, we'll break the loop
            break;
        }
    }
}

```

But remember that you have to pass `challengeNumber -2` to that function since our challenges start from 2.

Reply:
ok I'll do & test it